tests/integration_tests/tests/agentless_tests/test_blueprint_upload_autoupload_plugins.py

```diff
@@ -0,0 +1,193 @@
+
+    def test_blueprint_upload_autoupload_plugins_newer_version(self):
+        self._upload_and_verify_blueprint(
+            'bp1_os2',
+            'blueprint_with_plugins_from_catalog_os2.yaml')
+
+        plugin_versions = sorted(p.package_version for p in \
+                   self.client.plugins.list())
+        self.assertEqual(len(plugin_versions), 1)
+        self.assertRegex(plugin_versions[0], "2.*")
+
+        self._upload_and_verify_blueprint(
+            'bp1_os3',
+            'blueprint_with_plugins_from_catalog_os3.yaml')
+
+        plugin_versions = sorted(p.package_version for p in \
+                   self.client.plugins.list())
+        self.assertEqual(len(plugin_versions), 2)
+        self.assertRegex(plugin_versions[0], "2.*")
+        self.assertGreater(parse_version(plugin_versions[1]),
+                           parse_version("3"))
+
+    def test_blueprint_upload_autoupload_plugins_older_version(self):
+        self._upload_and_verify_blueprint(
+            'bp1_os3',
+            'blueprint_with_plugins_from_catalog_os3.yaml')
+
+        plugin_versions = sorted(p.package_version for p in \
+                   self.client.plugins.list())
+        self.assertEqual(len(plugin_versions), 1)
+        self.assertGreater(parse_version(plugin_versions[0]),
+                           parse_version("3"))
+
+        self._upload_and_verify_blueprint(
+            'bp1_os2',
+            'blueprint_with_plugins_from_catalog_os2.yaml')
+
+        plugin_versions = sorted(p.package_version for p in \
+                   self.client.plugins.list())
+        self.assertEqual(len(plugin_versions), 2)
+        self.assertRegex(plugin_versions[0], "2.*")
+        self.assertGreater(parse_version(plugin_versions[1]),
+                           parse_version("3"))
+
+    def test_blueprint_upload_autoupload_plugins_network_error(self):
+        from integration_tests.framework import utils, docker
+
+        # block github in /etc/hosts
+        docker.execute(
+            self.env.container_id,
+            'sh -c "sudo cat /etc/hosts > /tmp/old_hosts"')
+        docker.execute(
+            self.env.container_id,
+            'sh -c "echo 127.0.0.1  github.com > /tmp/new_hosts"')
+        docker.execute(
+            self.env.container_id,
+            'sh -c "cat /tmp/new_hosts > /etc/hosts"')
+
+        blueprint_id = 'bp_os3_no_github'
+        blueprint_filename = 'blueprint_with_plugins_from_catalog_os3.yaml'
+        self.assertRaisesRegexp(
+            CloudifyClientError,
+            'Couldn\'t download plugin from',
+            self.client.blueprints.upload,
+            resource('dsl/{}'.format(blueprint_filename)),
+            entity_id=blueprint_id)
+        blueprint = self.client.blueprints.get(blueprint_id)
+        self.assertEqual(blueprint['state'], BlueprintUploadState.INVALID)
+
+        docker.execute(
+            self.env.container_id,
+            'sh -c "cat /tmp/old_hosts > /etc/hosts"')
```

Comment:
absolutely not. This is going to break and leak into other tests. It's going to require access to a file that it normally wouldn't need to. If you want to test this, mock out requests and test this in restservice tests.